code/modules/reagents/reagent_dispenser.dm

```diff
@@ -118,6 +128,59 @@
 /obj/structure/reagent_dispensers/fueltank/fire_act()
 	blob_act() //saving a few lines of copypasta
 
+/obj/structure/reagent_dispensers/compostbin
+	name = "compost tank"
+	desc = "A device that mulches up unwanted produce into usable fertiliser."
+	icon = 'icons/obj/objects.dmi'
+	icon_state = "comptank"
+	amount_per_transfer_from_this = 30
+	var/list/validCompostTypepaths = list(/obj/item/weapon/reagent_containers/food/snacks/grown, /obj/item/seeds, /obj/item/weapon/reagent_containers/food/snacks/grown/mushroom)
+
+/obj/structure/reagent_dispensers/compostbin/attackby(obj/item/weapon/W as obj, mob/user as mob)
+	var/load = 1
+	var/addAmt = 0
+	if(is_type_in_list(W, validCompostTypepaths))
+		if(istype(W,/obj/item/seeds))
+			addAmt = 2
+		else
+			var/obj/item/weapon/reagent_containers/food/snacks/grown/G = W
+			addAmt = G.compost_value
+	else
+		load = 0
+
+	if(load)
+		user << "<span class='notice'>[src] mulches up [W].</span>"
+		playsound(src.loc, 'sound/effects/blobattack.ogg', 50, 1)
+		user.unEquip(W)
+		if(addAmt)
+			reagents.add_reagent("compost",addAmt)
+		qdel(W)
+		return
+	else ..()
+
+/obj/structure/reagent_dispensers/compostbin/MouseDrop_T(atom/movable/O as mob|obj, mob/user as mob)
+	if(is_type_in_list(O, validCompostTypepaths))
+		user.visible_message("<span class='notice'>[user] begins quickly stuffing items into [src]!</span>")
+		var/staystill = get_turf(user)
+		for(var/obj/item/P in staystill)

```

Comment:
Should probably do a is_type_in_list(P, validCompostTypepaths)) here as well for safety in case some numbskull adds more to the typepathes without handling lower down